### PR TITLE
Clean up a bit, add container building

### PR DESCRIPTION
**Pull request by exaexa**
shamelessly reused from COBREXA.jl repo. :]

### CODE REVIEW

**Comment by exaexa** [comment]
PS we shouldn't merge this before I test the container deployment on cobrexa. :D

File: gigasom.def [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    mkdir -p "${GIT_WORK_TREE}"
+    git checkout -f
+
+%post
+    export JULIA_DEPOT_PATH=/user/.julia
+    export PATH=/opt/julia/bin:$PATH
+
+    cd project
+    julia --project -e 'import Pkg; Pkg.add("Tulip"); Pkg.instantiate(); Pkg.build(); Pkg.precompile();'
+
```
Comment (laurentheirendt) [inline gigasom.def:16]: I think Tulip is not required

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -1,34 +1,186 @@
...
+apptainer-release:
+  stage: assets
+  script:
+    - |
+      # build the container
+      alias apptainer=singularity
+      apptainer build gigasom.sif gigasom.def
+    - |
+      # push to GHCR
+      alias apptainer=singularity
```
Comment (laurentheirendt) [inline .gitlab-ci.yml:155]: best feature ever

Comment (exaexa) [reply .gitlab-ci.yml:155]: I hope they don't remove it in the next version. :D :D

File: gigasom.def (hunk 1 above)
Comment (exaexa) [reply gigasom.def:16]: oh noes, I was actually pretty sure I was removing it from everywhere. good catch.

**Comment by laurentheirendt** [comment]
looks good to me 👍🏼

**Pull request merged**